src/manifolds/GrassmannProjector.jl

```diff
@@ -0,0 +1,267 @@
+
+@doc raw"""
+    representation_size(M::Grassmann{n,k}, p::ProjectorPoint)
+
+Return the represenation size or matrix dimension of a point on the [`Grassmann`](@ref)
+`M` when using [`ProjectorPoint`](@ref)s, i.e. ``(n,n)``.
+"""
+@generated representation_size(::Grassmann{n,k}, p::ProjectorPoint) where {n,k} = (n, n)
+
+@doc raw"""
+    canonical_project!(M::Grassmann{n,k}, q::ProjectorPoint, p)
+
+Compute the canonical projection ``π(p)`` from the [`Stiefel`](@ref) manifold onto the [`Grassmann`](@ref)
+manifold when represented as [`ProjectorPoint`](@ref), i.e.
+
+```math
+    π^{\mathrm{SG}}(p) = pp^{\mathrm{T}}
+```
+"""
+function canonical_project!(::Grassmann{n,k}, q::ProjectorPoint, p) where {n,k}
+    q.value .= p * p'
+    return q
+end
+function canonical_project!(
+    M::Grassmann{n,k},
+    q::ProjectorPoint,
+    p::StiefelPoint,
+) where {n,k}
+    return canonical_project!(M, q, p.value)
+end
+function allocate_result(
+    ::Grassmann{n,k},
+    ::typeof(canonical_project),
+    p::StiefelPoint,
+) where {n,k}
+    return ProjectorPoint(allocate(p.value, (n, n)))
+end
+
+@doc raw"""
+    canonical_project!(M::Grassmann{n,k}, q::ProjectorPoint, p)
+
+Compute the canonical projection ``π(p)`` from the [`Stiefel`](@ref) manifold onto the [`Grassmann`](@ref)
+manifold when represented as [`ProjectorPoint`](@ref), i.e.
+
+```math
+    Dπ^{\mathrm{SG}}(p)[X] = Xp^{\mathrm{T}} + pX^{\mathrm{T}}
+```
+"""
+function differential_canonical_project!(
+    ::Grassmann{n,k},
+    Y::ProjectorTVector,
+    p,
+    X,
+) where {n,k}
+    copyto!(Y.value, X * p' + p * X')
+    return Y
+end
+function differential_canonical_project!(
+    M::Grassmann{n,k},
+    Y::ProjectorTVector,
+    p::StiefelPoint,
+    X::StiefelTVector,
+) where {n,k}
+    differential_canonical_project!(M, Y, p.value, X.value)
+    return Y
+end
+function allocate_result(
+    ::Grassmann{n,k},
+    ::typeof(differential_canonical_project),
+    p::StiefelPoint,
+    X::StiefelTVector,
+) where {n,k}
+    return ProjectorTVector(allocate(p.value, (n, n)))
+end
+function allocate_result(
+    ::Grassmann{n,k},
+    ::typeof(differential_canonical_project),
+    p,
+    X,
+) where {n,k}
+    return ProjectorTVector(allocate(p, (n, n)))
+end
+
+@doc raw"""
+    exp(M::Grassmann, p::ProjectorPoint, X::ProjectorTVector)
+
+Compute the exponential map on the [`Grassmann`](@ref) as
+
+```math
+    \exp_pX = \operatorname{Exp}([X,p])p\operatorname{Exp}(-[X,p]),
+```
+where ``\operatorname{Exp}`` denotes the matrix exponential and ``[A,B] = AB-BA`` denotes the matrix commutator.
+
+For details, see Proposition 3.2 in [^BendokatZimmermannAbsil2020].
+"""
+exp(M::Grassmann, p::ProjectorPoint, X::ProjectorTVector)
+
+function exp!(::Grassmann, q::ProjectorPoint, p::ProjectorPoint, X::ProjectorTVector)
+    xppx = X.value * p.value - p.value * X.value
+    q.value .= exp(xppx) * p.value * exp(-xppx)
```

Comment:
```suggestion
    exp_xppx = exp(xppx)
    q.value .= exp_xppx * p.value / exp_xppx
```
since `exp(-xppx) = inv(exp(xppx))`.